server/src/main/java/org/opensearch/cluster/metadata/IndexAbstractionResolver.java

```diff
@@ -105,7 +106,7 @@ public List<String> resolveIndexAbstractions(
 
             // we always need to check for date math expressions
             final String dateMathName = indexNameExpressionResolver.resolveDateMathExpression(indexAbstraction);
-            if (dateMathName != indexAbstraction) {
+            if (!Objects.equals(dateMathName, indexAbstraction)) {
                 assert dateMathName.equals(indexAbstraction) == false;
```

Comment:
I'm wondering why do we even need this assertion. Why can remove this since that's the entry criteria in this block.